src/lib.rs

```diff
@@ -218,11 +278,15 @@ pub struct SNARKGens {
 impl SNARKGens {
   /// Constructs a new `SNARKGens` given the size of the R1CS statement
   pub fn new(num_cons: usize, num_vars: usize, num_inputs: usize, num_nz_entries: usize) -> Self {
-    let gens_r1cs_sat = R1CSGens::new(b"gens_r1cs_sat", num_cons, num_vars);
+    let mut vars_pad = 0;
```

Comment:
It seems like we are duplicating the logic to determine the number of padded variables in multiple places. Is there a way to avoid this?

Reply:
I made this a bit more clean, but at this point, we don't have an `Instance` in scope